emulator.c

```diff
@@ -812,6 +813,53 @@ update_sign_flag(i8080 *cpu, uint8_t result)
     }
 }
 
+// Interrupts
+
+// UPDATE GRAPHICS
+
+void
+update_graphics(i8080 *cpu, SDL_Surface *buffer, SDL_Surface *surface)
+{
+
+  uint32_t *screen_buff = buffer->pixels;
+
+  // Graphics data is rotated 90 degrees in memory counter-clockwise.  Reading
+  // byte by byte starting at 0x2400 we need to fill in the screen left to
+  // right, bottom to top.
+  int vram = 0x2400; // NOLINT
+  // Start at the left edge
+  for (int column = 0; column > SCREEN_WIDTH; column++)
+    {
+      // Start at bottom of screen, decrement by 8 since each bit is a pixel.
+      for (int row = SCREEN_HEIGHT; row > 0; row -= 8) // NOLINT
+        {
+          // Set each pixel based on bit value.
+          for (int pixel = 0; pixel < 8; pixel++) // NOLINT
+            {
+              // Calculate surface index.
+              int surf_index = (SCREEN_WIDTH * (row - pixel)) + column
+                               - (SCREEN_WIDTH - 1);
+
+              uint8_t cur_byte = cpu_read_mem(cpu, vram);
+
+              // Set pixel to on by changing color to white.
+              if (cur_byte & 1 << pixel)
```

Comment:
I don't understand what this is checking. Could you comment to explain or add parentheses? I can't tell what expected behavior is.

Reply:
I think this might just be messed up.  The logic is check each bit in the byte to see if the pixel is lit or not(since each bit is a pixel).  So you bitshift to the bit you want and then & 1 to check it.  set the pixel accordingly on the surface.